sdk/core/Azure.Core.TestFramework/src/TestEnvVar.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Azure.Core.TestFramework
+{
+    public class TestEnvVar : DisposableConfig
+    {
+        private static SemaphoreSlim _lock { get; } = new(1, 1);
+        public TestEnvVar(string name, string value) : base(name, value, _lock) { }
+        public TestEnvVar(Dictionary<string, string> values) : base(values, _lock) { }
+
+        internal override void SetValue(string name, string value)
+        {
+            _originalValues[name] = Environment.GetEnvironmentVariable(name);
+
+            CleanExistingEnvironmentVariables();
+
+            Environment.SetEnvironmentVariable(name, value as string);
+        }
+
+        internal override void SetValues(Dictionary<string, string> values)
+        {
+            foreach (var kvp in values)
+            {
+                _originalValues[kvp.Key] = kvp.Value;
+            }
+
+            CleanExistingEnvironmentVariables();
+
+            foreach (var kvp in values)
+            {
+                Environment.SetEnvironmentVariable(kvp.Key, kvp.Value as string);
+            }
+        }
+
+        internal override void InitValues()
+        {
+            // Common environment variables to be saved off for tests. Add more as needed
+            _originalValues["AZURE_USERNAME"] = Environment.GetEnvironmentVariable("AZURE_USERNAME");
```

Comment:
Hm, this is very identity-specific.

Reply:
I'd consider it just common variables that may already be set on the dev/test machine that need to be saved off and restored so that tests don't have to always protect against this.

Reply:
Should we do it for all "AZURE_*" then? I think we should only reset variables that can affect the current test.

Reply:
Let me try removing this altogether to see how many tests are affected in Identity and see how big of a pain it would be to add the affected variables to the affected tests.